packages/search-input/src/SearchInput/SearchInput.types.ts

```diff
@@ -62,6 +63,11 @@ interface BaseSearchInputProps
    * The placeholder text shown in the input field before the user begins typing.
    */
   placeholder?: string;
+
+  /**
+   * Callback fired on keydown event.
+   */
+  onKeyDown?: KeyboardEventHandler<HTMLDivElement>;
 }
```

Comment:
Note: line 53 is typed on the input element, line 59 on the form element, and this line on the div element. Feels pretty inconsistent. Aligning this is out of scope for this ticket, but this [ticket](https://jira.mongodb.org/browse/LG-3258) should help address this concern.